src/wp-includes/user.php

```diff
@@ -2221,8 +2221,20 @@ function wp_insert_user( $userdata ) {
 		$user_pass = ! empty( $userdata['user_pass'] ) ? $userdata['user_pass'] : $old_user_data->user_pass;
 	} else {
 		$update = false;
-		// Hash the password.
-		$user_pass = wp_hash_password( $userdata['user_pass'] );
+
+		if ( empty( $userdata['user_pass'] ) ) {
+			wp_trigger_error(
+				__FUNCTION__,
+				__( 'The user_pass field is required when creating a new user. A random password has been generated.' ),
+				E_USER_WARNING
+			);
+
+			// Generate and hash a random password
+			$user_pass = wp_hash_password( wp_generate_password( 32 ) );
+		} else {
+			// Hash the password.
+			$user_pass = wp_hash_password( $userdata['user_pass'] );
+		}
```

Comment:
This would look clean instead of nested `if..else`.
```suggestion
		$user_pass = ! empty( $user_pass['user_pass'] ) ? wp_hash_password( $user_pass['user_pass'] ) : wp_hash_password( wp_generate_password( 32 ) );
		
		if ( empty( $userdata['user_pass'] ) ) {
			wp_trigger_error(
				__FUNCTION__,
				__( 'The user_pass field is required when creating a new user. A random password has been generated.' ),
				E_USER_WARNING
			);
		}
```